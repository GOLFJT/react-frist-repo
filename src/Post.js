import React, { Component } from 'react'
import { 
 FormBox,
 Button,
 TextPost
} from './MainStyle'

class Post extends Component {

  render() {
    let postItems = this.props.postItems
    return (
      <FormBox horizontal>
        {postItems.map((post, key) => (
            <FormBox listPost key={key+1}>
               <TextPost>{key+1}. {post}</TextPost>
               <Button delete onClick={() => this.props.deleteItem(key)}>ลบ</Button>
            </FormBox>
        ))}
      </FormBox>
    )
  }
}

export default Post
