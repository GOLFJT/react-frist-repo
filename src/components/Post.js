import React, { Component } from 'react'
import { 
 FormBox,
 Button
} from './../styles/MainStyle'

class Post extends Component {

  render() {
    let postItems = this.props.postItems
    return (
      <FormBox horizontal>
        {postItems.map((post, key) => (
            <FormBox listPost key={key+1}>
               <span>{key+1}. {post}</span>
               <Button delete onClick={() => this.props.deleteItem(key)}>ลบ</Button>
            </FormBox>
        ))}
      </FormBox>
    )
  }
}

export default Post
