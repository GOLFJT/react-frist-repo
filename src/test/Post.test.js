import React from 'react'
import Post from './../components/Post.js'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
const prop = ['prop']  
const jsonPost = renderer.create(<Post postItems={ [...prop] } deleteItem={()=>{}} />).toJSON()
  expect(jsonPost).toMatchSnapshot()
})

