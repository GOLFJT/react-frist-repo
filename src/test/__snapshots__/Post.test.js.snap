// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-EHOje hkTeaZ"
>
  <div
    className="sc-EHOje ghyTZr"
  >
    <span>
      1
      . 
      prop
    </span>
    <button
      className="sc-bxivhb gnZhbm"
      onClick={[Function]}
    >
      ลบ
    </button>
  </div>
</div>
`;
